logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -138,9 +149,13 @@ def worker_multi_receive(events)
   def do_close
     @logger.debug("closing output delegator", :klass => self)
 
-    @worker_count.times do
-      worker = @worker_queue.pop
-      worker.do_close
+    if @threadsafe
+      @workers.each(&:do_close)
+    else
+      worker_count.times do
```

Comment:
is it possible that there is one or more workers popped from the list when do_close is called?
If threadsafe is false, `worker_count` is called to know how many pops/do_close we need to do. but worker_count computes @workers.size on each call